blog/_posts/2021-08-20-Mitchell-Johnson_Jamie-Schmidt.md

```diff
@@ -0,0 +1,397 @@
+  
+```cpp
+  
+  size_t Mutate(emp::Random & random) override { // receives a random number generator
+      // if we haven't configured the mutation sites yet, do so and set up the binomial distribution
+      if (mut_sites.GetSize() != data.size()) { 
+        mut_dist.Setup(mut_p, data.size());
+        mut_sites.resize(data.size());
+      }
+      // get the number of mutations
+      const size_t num_muts = mut_dist.PickRandom(random); 
+      
+
+      if (num_muts == 0) return 0;
+      if (num_muts == 1) {
+        const size_t pos = random.GetUInt(data.size());
+        data.Toggle(pos); // swaps the bit
+        return 1;
+      }
+
+      // Only remaining option is num_muts > 1.
+      mut_sites.Clear();
+      for (size_t i = 0; i < num_muts; i++) {
+        const size_t pos = random.GetUInt(data.size());
+        if (mut_sites[pos]) { --i; continue; }  // duplicate position; try again.
+        mut_sites.Set(pos); // sets each site of mutation to 1
+      }
+      data ^= mut_sites; // computes an exclusive or with the vector of bits and the mutation position vector
+
+      return num_muts;
+    }
+  
+```  
+
+Naturally, there are many more ways to mutate a Genome, particuarly if the Genome is a vector of integers or doubles.  If you want to make a Genome that mutates in a particular fashion, this is the way you'll have to go about it. 
+
+## DynamicOrg / author: mitchell and jamie
+In MABE2, an Organism can be thought of as a box which contains some number of Genomes and some  number of Brains.  An Organism has three main functions: 
+1. An Organism needs to be able to copy itself to form one or more children.
+2. An Organism needs to be able to mutate itself in some way for evolution to happen.
+3. An Organism needs to return some kind of object which consists of all the data the Evaluator needs. 
+
+Of course, different Organisms can handle things differently, so for example an Organism in MABE (call it a BitOrg) could consist entirely of one TypedGenome<bool>, which is a Genome that is composed entirely of bits. It can copy that whenever it reproduces, it can flip some random number of bits every time it mutates, and it can simply return the bits of that Genome in a BitVector (a specialized vector of bits which Empirical uses in place of std::BitMaps and std::vector<bool>s).
+  
+But what's a DynamicOrg?  Well, a DynamicOrg is an Organism that is nearly totally built by the config file. Unlike our previous example, where the BitOrg knew it had a Genome of bits and how to mutate and return it, a DynamicOrg can't.  Ideally, you'd be able to specify what Genomes and Brains were in this DynamicOrg, and that would be it. Specifying Genomes is relatively easy, since they're all completely self-contained, but the Brains pose a more significant challenge, which is why we'll be discussing them in the section after this.
```

Comment:
Why can't the ideal be achieved? What restrictions? I would move this to the other section if that's where this is actually covered, otherwise the reader is left hanging.

Also next paragraph starts with opposite tone making the case for DynamicOrg being less restrictive so it's odd to first mention there were significant challenges that will be addressed later.

Reply:
The ideal was achieved, it was just a weirdly worded line.